crates/torii/core/src/executor.rs

```diff
@@ -52,28 +53,29 @@ pub enum QueryType {
     Other,
 }
 
-impl QueryQueue {
-    pub fn new(pool: Pool<Sqlite>) -> Self {
-        QueryQueue { pool, queue: VecDeque::new(), publish_queue: VecDeque::new() }
-    }
+pub struct TxExecutor {
+    pool: Pool<Sqlite>,
+    rx: Receiver<QueryMessage>,
+}
 
-    pub fn enqueue<S: Into<String>>(
-        &mut self,
-        statement: S,
-        arguments: Vec<Argument>,
-        query_type: QueryType,
-    ) {
-        self.queue.push_back((statement.into(), arguments, query_type));
-    }
+pub struct QueryMessage {
+    statement: String,
+    arguments: Vec<Argument>,
+    query_type: QueryType,
+}
 
-    pub fn push_publish(&mut self, value: BrokerMessage) {
-        self.publish_queue.push_back(value);
+impl TxExecutor {
+    pub fn new(pool: Pool<Sqlite>) -> (Self, Sender<QueryMessage>) {
+        let (tx, rx) = channel(100); // Adjust buffer size as needed
```

Comment:
If this `100` is directly related to the max connection, we may have it configurable as input param?